ruby/command-t/lib/command-t/vim.rb

```diff
@@ -81,12 +81,12 @@ def wildignore_to_regexp(str)
           elsif pattern.match(%r{\A\*\.([^*]+)\z})
             # *.something (match file with extension at any level)
             '\.' + Regexp.escape($~[1]) + '\z'
+          elsif pattern.match(%r{\A\*/([^/]+)/\*\z})
+            # */something/* (match directories at any level)
+            '(\A|/)' + Regexp.escape($~[1]) + '/.*(/|\z)'
```

Comment:
I have a feeling this should work as it was before: ie. with `'(/|\z)'` at the end instead of `'/.*(/|\z)'`.

That is, the old version would have accepted "/dist/{literally anything}" or "dist{end of the string}". Your alternative `/.*(/|\z)` is equivalent to `/.*` because `.*` will greedily consume anything, and will therefore accept "dist/{literally anything}" too, but not "dist{end of string}".

If what I drew [in the table in the docs](https://github.com/wincent/command-t/blob/915048e6aed264dd7bf68e29ba485ade4c44f845/doc/command-t.txt#L1120-L1135) is right, then perhaps accepting "dist{end of string}" is not desirable, so your version is better. But I think we should go for the simpler and more intention-revealing `/.*` — or better still perhaps, `/.+` — in that case. So:

    '(\A|/)' + Regexp.escape($~[1]) + '/.+'

or:

    '(\A|/)' + Regexp.escape($~[1]) + '/.+\z'